Regression with fabio-0.9

```
lintaillefer:~/workspace/pyFAI % python3 run_tests.py pyFAI.test.test_integrate_app.suite
INFO:root:Generating grammar tables from /usr/lib/python3.5/lib2to3/Grammar.txt
INFO:root:Generating grammar tables from /usr/lib/python3.5/lib2to3/PatternGrammar.txt
INFO:pyFAI.setup:Use setuptools
INFO:pyFAI.setup:Use setuptools.setup
WARNING:run_tests:Building pyFAI to /users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5
INFO:root:Generating grammar tables from /usr/lib/python3.5/lib2to3/Grammar.txt
INFO:root:Generating grammar tables from /usr/lib/python3.5/lib2to3/PatternGrammar.txt
INFO:pyFAI.setup:Use setuptools
running build
running build_py
running build_ext
WARNING:run_tests:Patched sys.path, added: '/users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5'
WARNING:py.warnings:/usr/lib/python3.5/importlib/_bootstrap_external.py:423: ImportWarning: Not importing directory /usr/lib/python3/dist-packages/sphinxcontrib: missing __init__
  _warnings.warn(msg.format(portions[0]), ImportWarning)

WARNING:run_tests:Test pyFAI 0.18.0-dev1 from /users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5/pyFAI
.........F
Stdout:
                                                           

Stderr:
WARNING:pyFAI.worker:Wavelength is in meter... unlikely value 1
INFO:pyFAI.worker:Detector Detector	 Spline= None	 PixelSize= 1.720e-04, 1.720e-04 m
Wavelength= 1.000000e+00m
SampleDetDist= 1.583231e+00m	PONI= 3.341702e-02, 4.122778e-02m	rot1=0.006487  rot2= 0.007558  rot3= 0.000000 rad
DirectBeamDist= 1583.310mm	Center: x=179.981, y=263.859 pix	Tilt=0.571 deg  tiltPlanRotation= 130.640 deg
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
INFO:pyFAI:[First frame] Preprocessing time: 1ms
INFO:pyFAI:[Per frames] Reading time: 0ms; Processing time: 1ms
INFO:pyFAI:[Total] Reading time: 0.001s; Processing time: 0.003s
INFO:pyFAI:Execution done in 0.100s !
WARNING:py.warnings:/users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5/pyFAI/test/test_integrate_app.py:195: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals(result.shape[0], 4)

.................
======================================================================
FAIL: test_integrate_with_hdf5_monitor (pyFAI.test.test_integrate_app.TestIntegrateApp)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5/pyFAI/test/test_integrate_app.py", line 197, in test_integrate_with_hdf5_monitor
    numpy.testing.assert_almost_equal(result[iframe], expected, decimal=1)
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 573, in assert_almost_equal
    return assert_array_almost_equal(actual, desired, decimal, err_msg)
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 979, in assert_array_almost_equal
    precision=decimal)
  File "/usr/lib/python3/dist-packages/numpy/testing/utils.py", line 796, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 1 decimals

(mismatch 66.66666666666666%)
 x: array([ 33.9,  52. ,   0. ], dtype=float32)
 y: array([ 17.,  26.,   0.])

Stdout:
                                                           

Stderr:
WARNING:pyFAI.worker:Wavelength is in meter... unlikely value 1
INFO:pyFAI.worker:Detector Detector	 Spline= None	 PixelSize= 1.720e-04, 1.720e-04 m
Wavelength= 1.000000e+00m
SampleDetDist= 1.583231e+00m	PONI= 3.341702e-02, 4.122778e-02m	rot1=0.006487  rot2= 0.007558  rot3= 0.000000 rad
DirectBeamDist= 1583.310mm	Center: x=179.981, y=263.859 pix	Tilt=0.571 deg  tiltPlanRotation= 130.640 deg
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
WARNING:pyFAI:Fail to load monitor. No normalization applied. File format '<class 'fabio.hdf5image.Hdf5Frame'>' unsupported
INFO:pyFAI:[First frame] Preprocessing time: 1ms
INFO:pyFAI:[Per frames] Reading time: 0ms; Processing time: 1ms
INFO:pyFAI:[Total] Reading time: 0.001s; Processing time: 0.003s
INFO:pyFAI:Execution done in 0.100s !
WARNING:py.warnings:/users/kieffer/workspace-400/pyFAI/build/lib.linux-x86_64-3.5/pyFAI/test/test_integrate_app.py:195: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals(result.shape[0], 4)

```

--- a/pyFAI/utils/header_utils.py
+++ b/pyFAI/utils/header_utils.py
@@ -35,7 +35,7 @@ __author__ = "Valentin Valls"
 __contact__ = "valentin.valls@esrf.eu"
 __license__ = "MIT"
 __copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
-__date__ = "19/02/2019"
+__date__ = "08/03/2019"
 
 
 import logging
@@ -158,11 +158,19 @@ def get_monitor_value(image, monitor_key):
     if monitor_key is None:
         return Exception("No monitor defined")
 
-    if isinstance(image, fabio.edfimage.EdfImage):
-        return _get_monitor_value_from_edf(image, monitor_key)
-    elif isinstance(image, fabio.numpyimage.NumpyImage):
-        return _get_monitor_value_from_edf(image, monitor_key)
-    elif isinstance(image, fabio.hdf5image.Hdf5Image):
-        return _get_monitor_value_from_hdf5(image, monitor_key)
+    if fabio.version_info[0:2] < (0, 9):
+        if isinstance(image, fabio.edfimage.EdfImage):
+            return _get_monitor_value_from_edf(image, monitor_key)
+        elif isinstance(image, fabio.numpyimage.NumpyImage):
+            return _get_monitor_value_from_edf(image, monitor_key)
+        elif isinstance(image, fabio.hdf5image.Hdf5Image):
+            return _get_monitor_value_from_hdf5(image, monitor_key)
     else:
-        raise Exception("File format '%s' unsupported" % type(image))
+        if isinstance(image, (fabio.edfimage.EdfImage, fabio.edfimage.EdfFrame)):
+            return _get_monitor_value_from_edf(image, monitor_key)
+        elif isinstance(image, fabio.numpyimage.NumpyImage):
+            return _get_monitor_value_from_edf(image, monitor_key)
+        elif isinstance(image, (fabio.hdf5image.Hdf5Image, fabio.hdf5image.Hdf5Frame)):
+            return _get_monitor_value_from_hdf5(image, monitor_key)
+
+    raise Exception("File format '%s' unsupported" % type(image))

--- a/pyFAI/utils/header_utils.py
+++ b/pyFAI/utils/header_utils.py
@@ -159,6 +159,7 @@ def get_monitor_value(image, monitor_key):
         return Exception("No monitor defined")
 
     if fabio.version_info[0:2] < (0, 9):
+        # FIXME: Remove this dead code by upgrading the dependancy to fabio>=0.9
         if isinstance(image, fabio.edfimage.EdfImage):
             return _get_monitor_value_from_edf(image, monitor_key)
         elif isinstance(image, fabio.numpyimage.NumpyImage):

